### ISSUE TITLE
Not build from XCode 11.3 (11C29)

### ISSUE THREAD

**Issue by antondevs** [leetal/ios-cmake#71]
-- Merging integrated CMake 3.14+ iOS,tvOS,watchOS,macOS toolchain(s) with this toolchain!
-- Building with Xcode version:
-- Configuring iphoneos build for platform: OS, architecture(s): armv7;armv7s;arm64
-- Using SDK:  for platform: OS
-- Using the default min-version since DEPLOYMENT_TARGET not provided!
-- Enabling bitcode support by default. ENABLE_BITCODE not provided!
-- Enabling ARC support by default. ENABLE_ARC not provided!
-- Hiding symbols visibility by default. ENABLE_VISIBILITY not provided!
-- Using NON-strict compiler checks by default. ENABLE_STRICT_TRY_COMPILE not provided!
CMake Error at toolchain/ios.toolchain.cmake:395 (get_filename_component):
  get_filename_component called with incorrect number of arguments
Call Stack (most recent call first):
  /usr/local/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:4 (project)

CMake Error at toolchain/ios.toolchain.cmake:396 (get_filename_component):
  get_filename_component called with incorrect number of arguments
Call Stack (most recent call first):
  /usr/local/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:4 (project)

CMake Error at toolchain/ios.toolchain.cmake:399 (message):
  Invalid CMAKE_DEVELOPER_ROOT: does not exist.
Call Stack (most recent call first):
  /usr/local/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:4 (project)

CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
make: *** No targets specified and no makefile found.  Stop.

On previous version XCode - build is ok. How to fix it?

**Comment by antondevs** [comment]
cmake ../../ -DCMAKE_TOOLCHAIN_FILE=../../toolchain/ios.toolchain.cmake -DPLATFORM=OS -DCMAKE_BUILD_TYPE=Release

**Comment by leetal** [comment]
I will look on this today. I have not looked into all the changes kiteware have made in 3.16 yet, so I'll dig right into it during the day.

**Comment by leetal** [comment]
Is this still an issue for you? I have cloned you lib and can build it locally without any issues? Stupid question perhaps, but have you tried cleaning the build dir after upgrade of Xcode?

**Comment by leetal** [comment]
Closing due to inactivity. Please comment if the issue still exists.

**Comment by StephenCWills** [comment]
FYI, we encountered this same issue with one of our users who was trying to build DevilutionX for iOS...
`get_filename_component called with incorrect number of arguments`

Which pointed to line 550 in our case...
https://github.com/diasurgical/devilutionX/blob/35e93366f6bb8bc641b58927aa763957fd0d501e/CMake/platforms/ios.toolchain.cmake#L550

We investigated a bit and found the issue was due to an error returned by `xcodebuild`. When the user ran `xcodebuild --help`, they received the following error message:
`tool 'xcodebuild' requires XCode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance`

We found this post on StackOverflow which suggests navigating to `XCode > Preferences > Locations > Command Line Tools` and selecting an option from the dropdown. Following those instructions and running a clean build is what fixed the issue.
https://stackoverflow.com/questions/55939354/tool-xcodebuild-requires-xcode-but-active-developer-directory-library-devel/72115137#72115137

Probably what's happening is that the `xcodebuild` error message is written to stderr, and so the version is populated with an empty string.
https://github.com/leetal/ios-cmake/blob/ad96a372b168930c2a1ff9455e1a9ccb13021617/ios.toolchain.cmake#L601-L604
